src/main/groovy/com/ullink/Msbuild.groovy

```diff
@@ -122,11 +123,15 @@ class Msbuild extends ConventionTask {
     }
 
     boolean resolveProject() {
-        if (projectParsed == null) {
+        if (projectParsed == null && parseProject == true) {
             if (isSolutionBuild()) {
                 def result = parseProjectFile(getSolutionFile())
                 allProjects = result.collectEntries { [it.key, new ProjectFileParser(msbuild: this, eval: it.value)] }
                 projectParsed = allProjects[getProjectName()]
+                if (projectParsed == null) {
+                    parseProject = false
```

Comment:
True. What if we have getProjects just return allProjects. If resolveProject works then allProjects will be set, if something failed it will still be null.